libcodechecker/server/client_db_access_handler.py

```diff
@@ -1148,40 +1358,201 @@ def getDiffResultTypes(self,
+
+        run_id = self.addCheckerRun(command,
+                                    name,
+                                    version,
+                                    force)
+
+        # Handle skip list.
+        for skip_handler in skip_handlers:
+            if not self.addSkipPath(run_id, skip_handler.get_skiplist()):
+                LOG.debug("Adding skip path failed!")
+
+        # Processing PList files.
+        _, _, report_files = next(os.walk(report_dir), ([], [], []))
+        for f in report_files:
+            if not f.endswith('.plist'):
+                continue
+
+            LOG.debug("Parsing input file '" + f + "'")
+
+            try:
+                files, reports = plist_parser.parse_plist(
+                    os.path.join(report_dir, f))
+            except Exception as ex:
+                LOG.error('Parsing the plist failed: ' + str(ex))
+                continue
+
+            file_ids = OrderedDict()
+            # Store content of file to the server if needed.
+            for file_name in files:
+                file_ids[file_name] = file_path_to_id[file_name]
+
+            # Store report.
+            for report in reports:
+                last_report_event = report.bug_path[-1]
+                sp_handler = suppress_handler.SourceSuppressHandler(
+                    files[last_report_event['location']['file']],
+                    last_report_event['location']['line'],
+                    report.main['issue_hash_content_of_line_in_context'],
+                    report.main['check_name'])
+
+                supp = sp_handler.get_suppressed()
+                if supp:
+                    self.addSuppressBug(run_id, [
+                        shared.ttypes.SuppressBugData(*supp)])
+
+                LOG.debug("Storing check results to the database.")
+
+                checker_name = report.main['check_name']
+                context = generic_package_context.get_context()
+                severity_name = context.severity_map.get(checker_name,
+                                                         'UNSPECIFIED')
+                severity = \
+                    shared.ttypes.Severity._NAMES_TO_VALUES[severity_name]
+
+                bug_paths, bug_events = \
+                    store_handler.collect_paths_events(report, file_ids)
+
+                LOG.debug("Storing report")
+                report_id = self.addReport(
+                    run_id,
+                    file_ids[files[report.main['location']['file']]],
+                    report.main['issue_hash_content_of_line_in_context'],
+                    report.main['description'],
+                    bug_paths,
+                    bug_events,
+                    checker_name,
+                    report.main['category'],
+                    report.main['type'],
+                    severity)
+
+                LOG.debug("Storing done for report " + str(report_id))
+
+        if len(check_durations) > 0:
```

Comment:
Is this true that the duration of checks is the duration of the run? Shouldn't we count the duration of store as well?